specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-04-01-preview/apimauthorizationproviders.json

```diff
@@ -0,0 +1,805 @@
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The response body contains the specified authorization provider entity. No secrets included",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/AuthorizationProviderContract"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "AuthorizationProvider"
+        ],
+        "operationId": "AuthorizationProvider_CreateOrUpdate",
+        "description": "Creates or updates authorization provider.",
+        "x-ms-examples": {
+          "ApiManagementCreateAuthorizationProviderAADAuthCode": {
+            "$ref": "./examples/ApiManagementCreateAuthorizationProviderAADAuthCode.json"
+          },
+          "ApiManagementCreateAuthorizationProviderAADClientCred": {
+            "$ref": "./examples/ApiManagementCreateAuthorizationProviderAADClientCred.json"
+          },
+          "ApiManagementCreateAuthorizationProviderGenericOAuth2": {
+            "$ref": "./examples/ApiManagementCreateAuthorizationProviderGenericOAuth2.json"
+          },
+          "ApiManagementCreateAuthorizationProviderOOBGoogle": {
+            "$ref": "./examples/ApiManagementCreateAuthorizationProviderOOBGoogle.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/AuthorizationProviderIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./definitions.json#/definitions/AuthorizationProviderContract"
+            },
+            "description": "Create parameters."
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/IfMatchOptionalParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
>" [](http://example.com/codeflow?start=10&length=1)

Do you only support Update? What about Create?? Create of a tracked resource should be able to return 201 Created in addition to 200 OK.

Reply:
@LFZ96 can you take a look?

Reply:
@TimLovellSmith We support Create AuthorizationProvider
Should I change this to a 201 response status instead of the 200?

Reply:
I'm thinking you probably want both 200 and 201 responses for all your PUT operations.

Reply:
@LFZ96 this is guideline @TimLovellSmith  is referring to

[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc015-put-apis-that-only-return-200](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc015-put-apis-that-only-return-200)